CI: Automatically apply pull request labels for generic PR actions

Lubrsi opened the pull request:
Generic PR actions include opening a PR, submit review comments, adding new commits, etc. This prevents the reviewer and PR submitter from having to manually bounce the labels back and forth in the general case.

This does not handle more subjective labels such as pr-is-blocked and pr-unclear. Unfortunately, there does not seem to be a GitHub Actions trigger for when a PR has merge conflicts.

Lubrsi reviewed (commented):
does this count as a comment? will you put the pr-waiting-for-author label on for me if it does count?

AtkinsSJ reviewed (changes requested):
Hmmmm. :thinking: :thinking: :thinking: :thinking: :thinking:

AtkinsSJ reviewed (commented):
This is a review, that's a comment

Lubrsi commented:
a comment from the submitter with no fix, you're gonna put on the `waiting-for-author` label aren't you?

Lubrsi reviewed (commented):
how about now?

AtkinsSJ commented on .github/workflows/label-pull-requests.yml:
```diff
@@ -0,0 +1,68 @@
+                return ["pr-waiting-for-author"];
+            }
+
+            function labelsForPullRequestConvertedToDraft() {
+                return [];
+            }
```
Oh no you did a thing, change it immediately!
And then reply to this comment.

Lubrsi replied:
oh no, fixed

Lubrsi replied:
anotha one

Lubrsi replied:
comment

Lubrsi commented on .github/workflows/label-pull-requests.yml:
```diff
@@ -0,0 +1,70 @@
+      - name: Label pull request
+        uses: actions/github-script@v6
+        with:
+          script: |
```
I make a comment. What do you apply?

alimpfard replied:
Oops, commented in the wrong box

Lubrsi replied:
comment

alimpfard commented:
bad PR, remove immediately

Lubrsi commented:
no u

Lubrsi commented:
comment

Lubrsi commented:
commmmmmmmmenttttttttt

Lubrsi commented:
comment!

Lubrsi committed:
CI: Automatically apply pull request labels for generic PR actions

Generic PR actions include opening a PR, submit review comments, adding
new commits, etc. This prevents the reviewer and PR submitter from
having to manually bounce the labels back and forth in the general
case. The reviewer also may not have permission to set labels, meaning
the reviewer won't be able to update the labels accordingly themselves.

This does not handle more subjective labels such as pr-is-blocked and
pr-unclear. Unfortunately, there does not seem to be a GitHub Actions
trigger for when a PR has merge conflicts, so the pr-has-conflicts
label cannot be automatically applied.

I couldn't figure out how to make this work for general PR comments
that are not review comments, so this has been left as a FIXME for now.
